### PR TITLE
siPhase2Clusters in RECOSIM event content

### PR DESCRIPTION
**Pull request by ebrondol**
This PR is done in order to allow the Validation tools used only for the tracking part to run on RECO events. The tracking MC association is not possible when the clusters information is not stored.

@makortel please check
@kpedro88 this is the second PR that will fix Validation for tracking. Good to have in pre8 if possible, as I mentioned in #14951 
@VinInn @rovere

### CODE REVIEW

**Comment by kpedro88** [comment]
@cmsbuild please test

File: RecoLocalTracker/Configuration/python/RecoLocalTracker_EventContent_cff.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
     'keep DetIdedmEDCollection_siPixelDigis_*_*',
     'keep *_siPixelClusters_*_*', 
     'keep *_siStripClusters_*_*',
+    'keep *_siPhase2Clusters_*_*',
```
Comment (makortel) [inline RecoLocalTracker/Configuration/python/RecoLocalTracker_EventContent_cff.py]: I believe these would be better added only with `phase2_tracker` era, see e.g. https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/RecoLocalMuon/Configuration/python/RecoLocalMuon_EventContent_cff.py#L31.

**Comment by makortel** [comment]
@ebrondol If you're going to make changes, I'd suggest to rebase to not to include the commit from #14951 since the two PRs are independent.

**Comment by ebrondol** [comment]
@makortel Thanks, this was a really good hint. 
About the rebasing, you are right. If you think my changes are fine, I can do another PR with just that (without merging yours).

**Comment by kpedro88** [comment]
@ebrondol you don't have to do another PR, just rebase this one.

**Comment by VinInn** [comment]
rebase?

**Comment by ebrondol** [comment]
yeah, and failed. The new PR is in #14982

**Pull request closed without merging**